tests/http-adapter.spec.js

```diff
@@ -86,7 +86,7 @@ describe('HttpAdapter', () => {
       httpAdapter.get(path2)
 
       // Tick microtask queue
-      await new Promise(setImmediate)
+      await new Promise(setTimeout)
```

Comment:
Non-blocker q: what helped you decide whether to use `setTimeout` instead of `setImmediate`?

Reply:
`setImmediate` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate) and `setTimeout` does the same thing for our purposes

Reply:
Good to know!